core/src/main/scala/org/ensime/core/AnalyzerManager.scala

```diff
@@ -98,25 +82,24 @@ class AnalyzerManager(
       config.projects.flatMap(_.libraryJars).toSet,
       config.projects.flatMap(_.librarySources).toSet,
       config.projects.flatMap(_.libraryDocs).toSet
-    ))
+    )
+  )
 
   override def preStart(): Unit = {
-    // legacy clients expect to see FullTypeCheckComlpeteEvent and AnalyzerReadyEvent on connection
-    broadcaster ! Broadcaster.Persist(AnalyzerReadyEvent)
-    broadcaster ! Broadcaster.Persist(FullTypeCheckCompleteEvent)
     // initialize loadedFiles
-    config.projects foreach (p => historyOfModule += (p -> LoadedFilesData(Map(), NotLoaded)))
+    config.projects foreach (p => historyOfModule += (p -> LoadedFilesData(Map.empty, NotLoaded)))
   }
 
   override def receive: Receive = ready
 
   private def ready: Receive = withLabel("ready") {
     case ReloadExistingFilesEvent =>
-      for {
-        (module, analyzer) <- analyzers
-      } analyzer forward ReloadExistingFilesEvent
       if (analyzers.isEmpty)
-        broadcaster ! AnalyzerReadyEvent
+        context.parent ! AnalyzerReadyEvent
```

Comment:
try to avoid using `context.parent` when sending messages. Try to use the sender or the `broadcaster` where possible. Baking in the `context.parent` implies that the actor hierarchy is rigid and usually indicates a design flaw.